api/handler.py

```diff
@@ -1,72 +1,85 @@
-        
-        transport_mode = vars.get("transport_mode")
+
+        LOGGER.info(
+            "estimate-co2 sourceIp=%s userAgent=%s, input=%s",
+            event["requestContext"]["identity"]["sourceIp"]
+            if event.get("requestContext")
+            else "?",
+            event["requestContext"]["identity"]["userAgent"]
+            if event.get("requestContext")
+            else "?",
+            params,
+        )
+
+        transport_mode = params.get("transport_mode")
         if isinstance(transport_mode, str):
             transport_mode = Mode[transport_mode.upper()]
 
-        distance_km = strtofloat(vars.get("distance_km"))
-        vehicle_occupancy = strtofloat(vars.get("vehicle_occupancy"))
-        origin_lat = strtofloat(vars.get("origin_lat"))
-        origin_lon = strtofloat(vars.get("origin_lon"))
-        destination_lat = strtofloat(vars.get("destination_lat"))
-        destination_lon = strtofloat(vars.get("destination_lon"))
+        distance_km = strtofloat(params.get("distance_km"))
+        vehicle_occupancy = strtofloat(params.get("vehicle_occupancy"))
+        origin_lat = strtofloat(params.get("origin_lat"))
+        origin_lon = strtofloat(params.get("origin_lon"))
+        destination_lat = strtofloat(params.get("destination_lat"))
+        destination_lon = strtofloat(params.get("destination_lon"))
 
         estimate = get_co2_estimate(
-            transport_mode = transport_mode,
-            distance_km = distance_km,
-            vehicle_occupancy = vehicle_occupancy, 
-            origin_lat = origin_lat, 
-            origin_lon = origin_lon,
-            destination_lat = destination_lat, 
-            destination_lon = destination_lon
+            transport_mode=transport_mode,
+            distance_km=distance_km,
+            vehicle_occupancy=vehicle_occupancy,
+            origin_lat=origin_lat,
+            origin_lon=origin_lon,
+            destination_lat=destination_lat,
+            destination_lon=destination_lon,
         )
 
         if estimate.get("error"):
             response = {
                 "statusCode": 400,
-                "body": json.dumps({
-                    "error:": estimate.get("error")
-                })
+                "body": json.dumps({"error:": estimate.get("error")}),
             }
             return response
 
         response = {
             "statusCode": 200,
-            "body": json.dumps({
-                "input": vars,
-                "result": estimate
-            })
+            "body": json.dumps({"input": params, "result": estimate}),
         }
         return response
-            
-            
-    except Exception as err:
-        logging.exception(f"Got exception processing request: {type(err)}: {err}")
+
+    except Exception:
+        LOGGER.exception("Got exception processing request: %r", event)
```

Comment:
When using `logging.exception()` it will automatically log the exception incl stack trace, so no need to include it in the error message.